docs/decisions/2025-04-02_scalar.md

```diff
@@ -0,0 +1,31 @@
+# 2025-04-02 Use Git Scalar for Clones
+
+- **status**: rejected
+- **date**: 2025-04-02
+- **authors**: Chaya Malik
+
+## Context and Problem Statement
+
+The Developer Productivity team evaluated using Git Scalar to optimize cloning operations for large repositories within Evergreen. This exploration was motivated by the potential for enhanced performance through improved clone speed and network efficiency, as documented in ticket DEVPROD-9990.
+
+## Integration Approach
+
+The integration of Git Scalar into our workflow involved a preliminary assessment of compatibility, particularly focusing on the installed Git version. Git Scalar required Git version 2.42.1 to support the `--no-src` option, which was essential for ensuring repositories were not placed in a `src` directory, thereby minimizing disruption to existing scripts. We upgraded our hosts to Git version 2.42.1 where possible, and moved away from using Windows' packaged versions of Git (or Apple's) as scalar was not bundled with it.
+
+Our eventual goal was to use scalar as the default across the board for all projects so that all clones done in evergreen through git.get_project become more efficient. However, to ensure that we don't accidentally break anything, we decided to do it in steps.
+
+We added a `use_scalar` option to git.get_project to use `scalar clone --no-src` instead of `git clone` if the git version on the host was compatible. We did not allow combining it with with clone_depth, shallow_clone, or recurse_submodules as they were not available with scalar.
+
+We also added a `use_scalar_full_clone` option which was same as use_scalar with the addition of --full-clone. Scalar initializes sparse-checkout by default, meaning only files in the top-level directory will be present. This option turned it off ensuring all files were present. We added this after discovering that some projects had tasks that relied on it for their workflow. We therefore couldn't beta test scalar without it.
```

Comment:
nit: \`use_scalar\` and \`--full-clone\` (with tildes) to be consistent. Same with the paragraph below.